rfcs/text/0013_background_sessions.md

```diff
@@ -0,0 +1,491 @@
+// Any follow up searches executed with this sessionId will be saved into this object as well.  
+const backgroundSession = await searchService.session.store();
+```
+
+# Motivation
+
+Kibana is great at providing fast results from large sets of "hot" data. However, there is an increasing number of use cases where users want to analyze large amounts of "colder" data (such as year-over-year reports, historical or audit data, batch queries, etc.). 
+
+For these cases, users run into two limitations: 
+ 1. Kibana has a default timeout of 30s per search. This is controlled by the `elasticsearch.requestTimeout` setting (originally intended to protect clusters from unintentional overload by a single query).
+ 2. Kibana cancels queries upon navigating away from an application, once again, as means of protecting clusters and reducing unnecessary load. 
+
+In 7.7, with the introduction of the `_async_search` API in Elasticsearch, we provided Kibana users a way to bypass the timeout, but users still need to remain on-screen for the entire duration of the search requests.
+
+The primary motivation of this RFC is to enable users to do the following without needing to keep Kibana open, or while moving onto other work inside Kibana:
+
+- Run long search requests (beyond 30 seconds)
+- View their status (complete/incomplete)
+- Cancel incomplete search requests
+- Retrieve completed search request results 
+
+# Detailed design
+
+Because a single view (such as a dashboard with multiple visualizations) can initiate multiple search requests, we need a way to associate the search requests together in a single entity. 
+
+We call this entity a `session`, and when a user decides that they want to continue running the search requests while moving onto other work, we will create a saved object corresponding with that specific `session`, persisting the *sessionId* along with a mapping of each *request's hash* to the *async ID* returned by Elasticsearch.
+
+## High Level Flow Charts
+
+### Client side search
+
+This diagram matches any case where `data.search` is called from the front end:
+
+![image](../images/background_sessions_client.png)
+
+### Server side search
+
+This case happens if the server is the one to invoke the `data.search` endpoint, for example with TSVB.
+
+![image](../images/background_sessions_server.png)
+
+## Data and Saved Objects
+
+### Background Session Status
+
+```ts
+export enum BackgroundSessionStatus {
+  Running,   // The session has at least one running search ID associated with it.
+  Done,      // All search IDs associated with this session have completed.
+  Error,     // At least one search ID associated with this session had an error. 
+  Expired,   // The session has expired. Associated search ID data was cleared from ES.
+}
+```
+
+### Saved Object Structure
+
+The saved object created for a background session will be scoped to a single space, and will be a `hidden` saved object
```

Comment:
If we're only concerned with these saved-objects showing up in the management listing, you don't have to make them `hidden: true`, you can just not declare the [management](https://github.com/elastic/kibana/blob/d666038c8f6767f6790cb78f29e0027ff73d83ee/src/core/server/saved_objects/types.ts#L233) key.

However, I do think that we'll want these to be `hidden` saved-objects for another reason. Is my understanding correct that we only want the user that created a background session to be able to CRUD their own background sessions? As far as I'm aware, this is only possible currently by declaring the saved-object as being `hidden` and writing a custom "Client" which performs its own authorization and filtering of these saved-objects.

Reply:
> However, I do think that we'll want these to be hidden saved-objects for another reason. Is my understanding correct that we only want the user that created a background session to be able to CRUD their own background sessions? As far as I'm aware, this is only possible currently by declaring the saved-object as being hidden and writing a custom "Client" which performs its own authorization and filtering of these saved-objects.

++ that's absolutely right. We discussed this [here](https://github.com/elastic/kibana/pull/73281/files/65043a75db131e3899baec907762a698352555e2#r477515658) during an earlier iteration of this RFC

Reply:
Yes, that's correct. We don't want these saved objects showing up in the general saved object management listing. We will have a separate, background-session-specific management view. The list will be filtered based on the `userId` described below, and will be managed through the `expire`/`extend`/etc. APIs described below.